src/bootstrap/suggest.rs

```diff
@@ -0,0 +1,142 @@
+use crate::builder::Builder;
+use glob::Pattern;
+use std::fmt::{self, Display};
+
+/// Suggests a list of possible `x.py` commands to run based on modified files in branch.
+pub fn suggest(build: &Builder<'_>, run: bool) {
+    // Find the commit the current branch split from master.
+    let merge_base = build
+        .config
+        .git()
+        .arg("merge-base")
+        .arg("HEAD")
+        .arg("origin/master")
+        .output()
+        .expect("failed to execute `git`");
+
+    // `git` shouldn't ever give us non-utf8 output.
+    let merge_base = String::from_utf8(merge_base.stdout).expect("`git` gave non-utf8 commit?!");
+
+    // Note that `merge_base` ends in a newline; it must be trimmed.
+    let modified_files = build
+        .config
+        .git()
+        .arg("diff")
+        .arg("--name-only")
+        .arg(merge_base.trim())
+        .output()
+        .expect("failed to execute `git`");
+
+    let modified_files =
+        String::from_utf8(modified_files.stdout).expect("`git` gave non-utf8 commit?!");
+
+    let modified_files: Vec<_> = modified_files.split_terminator('\n').collect();
+
+    println!("=== MODIFIED FILES ===");
+    for file in &modified_files {
+        println!("{}", file);
+    }
+    println!("======================\n");
```

Comment:
this is fine for debugging, but please remove it before unmarking the PR as a draft. you can hide it behind `builder.verbose` if you like.